Sources/Bond/Data Structures/TreeNode.swift

```diff
@@ -24,9 +24,16 @@
 
 import Foundation
 
+public protocol TreeNodeWithValueProtocol: RangeReplaceableTreeProtocol {
+    associatedtype Value
+    var value: Value { get set }
+    init(_ value: Value)
```

Comment:
It is absolutely necessary to have the initialiser be a requirement of conformance to the protocol? This is getting in the way of conforming existing types to this protocol.

Reply:
It would require some refactoring of Array2D to achieve that. 

Btw, is the underlying tree type you have a value or a reference type?

Reply:
It’s an NSObject so reference type.

Reply:
I was able to relax the constraints so that any type conforming to `TreeProtocol` can be bound to `NSOutlineView`. You no longer need to use `TreeNode`/`TreeArray`.

Reply:
I've refactored `Array2D` and simplified `TreeNodeWithValueProtocol`.